# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasession/MediaSessionController.h
# Commit: a008d3cebfdb
# Full Hash: a008d3cebfdb9d96ab32530b90180141575316a8
# Author: alwu <alwu@mozilla.com>
# Date: 2020-03-17 09:36:40
# Regressor Bug: 1582508
# File Overlap Count: 1
# Description:
#   Bug 1582508 - part4 : propagate declared playback state to media session controller. r=chunmin
#   
#   This patch includes the implementation of propagating declared playback state from the media session in the content process to the media session controller in the chrome process.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D66341
# ==============================================================================

diff -r a73415e9c006 -r a008d3cebfdb dom/media/mediasession/MediaSessionController.h
--- a/dom/media/mediasession/MediaSessionController.h	Tue Mar 17 00:03:44 2020 +0000
+++ b/dom/media/mediasession/MediaSessionController.h	Tue Mar 17 00:17:36 2020 +0000
@@ -82,6 +82,15 @@
     return mMetadataChangedEvent;
   }
 
+  // This method is used to update the declared playback state for the specific
+  // media session.
+  virtual void SetDeclaredPlaybackState(uint64_t aSessionContextId,
+                                        MediaSessionPlaybackState aState);
+
+  // Return the active media session's declared playback state. If we don't have
+  // active media session, we would return 'None'.
+  MediaSessionPlaybackState GetCurrentDeclaredPlaybackState() const;
+
  protected:
   ~MediaSessionController() = default;
   uint64_t mTopLevelBCId;